docs/guides/how_to_guides/creating_batches/how_to_load_a_pandas_dataframe_as_a_batch.rst

```diff
@@ -1,18 +1,145 @@
+
+        .. admonition:: Prerequisites: This how-to guide assumes you have already:
+
+            - :ref:`Set up a working deployment of Great Expectations <tutorials__getting_started>`
+            - Identified a pandas DataFrame that you would like to use as the data to validate.
+
+        Steps
+        -----
+
+        1. Configure a Datasource
+
+            Configure a :ref:`Datasource <reference__core_concepts__datasources>` using the :ref:`RuntimeDataConnector <reference__core_concepts__datasources>` to connect to your DataFrame. Since we are reading a pandas DataFrame, we use the PandasExecutionEngine. You can use ``runtime_keys`` to define what data you are able to attach as additional metadata to your DataFrame using the ``partition_request`` parameter (shown in step 3).
+
+            .. code-block:: yaml
+
+                insert_your_pandas_datasource_name_here:
+                  class_name: Datasource
+                  execution_engine:
+                    class_name: PandasExecutionEngine
+                  data_connectors:
+                    insert_your_runtime_data_connector_name_here:
+                      module_name: great_expectations.datasource.data_connector
+                      class_name: RuntimeDataConnector
+                      runtime_keys:
+                        - some_key_maybe_pipeline_stage
+                        - some_other_key_maybe_run_id
+
+
+        2. Obtain an Expectation Suite
+
+            .. code-block:: python
+
+                suite = context.get_expectation_suite(insert_your_expectation_suite_name_here)
+
+            Alternatively, if you have not already created a suite, you can do so now.
+
+            .. code-block:: python
+
+                suite = context.create_expectation_suite(insert_your_expectation_suite_name_here)
+
+        3. Construct a BatchRequest
+
+            We will create a BatchRequest and pass it our DataFrame via the ``batch_data`` argument.
+
+            Attributes inside the ``partition_request`` are optional - you can use them to attach additional metadata to your DataFrame. When configuring the Data Connector, you used ``runtime_keys`` to define which keys are allowed.
+
+            NOTE: for now, data_asset_name can only be set to this predefined string: ``“IN_MEMORY_DATA_ASSET”``. We will fix it very soon and will allow you to specify your own name.
+
+            .. code-block:: python
+
+                from great_expectations.core.batch import BatchRequest
+
+                batch_request = BatchRequest(
+                    datasource_name="insert_your_pandas_datasource_name_here",
+                    data_connector_name="insert_your_runtime_data_connector_name_here",
+                    batch_data=insert_your_dataframe_here,
+                    data_asset_name="IN_MEMORY_DATA_ASSET",
+                    partition_request={
```

Comment:
Not related to the review, but I think my main takeaway from this is that partition_request and runtime_keys should be optional since they really blow up the code.